# <center> Практическое задание к уроку №3

___

**1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):**  
   **а) среднее арифметическое,**  
   **б) среднее квадратичное отклонение,**  
   **с) смещенную и несмещенную оценки дисперсий для данной выборки.**

In [1]:
import numpy as np
from math import factorial

In [8]:
arr = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

Среднее арифметическое: $\displaystyle M(X)=\frac{1}{n}\sum^n{i=1}x_i$

In [9]:
def mean_value(array):
    return sum(array)/len(array)

In [10]:
print(f'M(X) = {mean_value(arr): .2f}')

M(X) =  65.30


In [16]:
np.mean(arr)

65.3

Среднее квадратичное отклонение: $\sigma=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}}$

In [21]:
def mean_square_deviation(array):
    square_dev = (array - mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)

In [23]:
print(f'SD = {mean_square_deviation(arr): .4f}')

SD =  30.8239


Смещенная оценка дисперсии: $\sigma^2\frac{\displaystyle\sum^n{i=1}(x_{i}-\bar x)^2}{n}$

In [30]:
def dispers(array, no_off=False):
    square_dev = (array - mean_value(array))**2
    return sum(square_dev)/(len(square_dev) - 1) if no_off else sum(square_dev)/len(square_dev)

In [31]:
print(f'D1 = {dispers(arr): .4f}\n')
print(f'D2 = {dispers(arr, True): .4f}')

D1 =  950.1100

D2 =  1000.1158


In [32]:
np.var(arr)

950.11

In [33]:
np.var(arr, ddof=1)

1000.1157894736842

___

#### Ответ:

а) $М(Х)= 65.30$;  
б) $\sigma = 30.8239$;  
в) $\sigma^2_{\text{смещ}}=950.11$, $\sigma^2_{\text{несмещ}}=1000.1158$.  

---

**2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

Пусть $A$ - искомое событие. Оно может произойти:
    
1. $A1$ из первой корзины ни одного мяча, из второй - все три
2. $A2$ из перовой корзины - 1 мяч, из второй - 2
3. $A3$ из первой корзины - 2 мяча, из второй - 1
    
Вероятность $A$, будет оцениваться как сумма всех событий.

___

Вероятность события $A1$: $P(A_1)=\frac{C^2_3}{C^2_8}\cdot\frac{C^3_5\cdot C^1_7}{C^4_{12}}$

In [41]:
def combinations(n, k):
    return factorial(n)/(factorial(k) * factorial(n - k))

In [42]:
P1 = (combinations(3, 2)/combinations(8, 2)) * ((combinations(5, 3) * combinations(7, 1))/combinations(12, 4))
print(f'P1 = {round(P1, 4)}')

P1 = 0.0152


Вероятность события $A2$: $P(A_2)=\frac{C^1_5\cdot C^1_3}{C^2_8}\cdot\frac{C^2_5\cdot C^2_7}{C^4_{12}}$

In [43]:
P2 = (combinations(5,1) * combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))

In [48]:
print(f'P2 = {round(P2, 4)}')

P2 = 0.2273


Вероятность события $A$: $P(A_3) = \frac{C^2_5}{C^2_8}\cdot\frac{C^1_5\cdot C^3_7}{C^4_{12}}$

In [47]:
P3 = (combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))

In [49]:
print(f'P3 = {round(P2, 4)}')

P3 = 0.2273


$P(A) = P(A_1) + P(A_2) + P(A_3)$

In [51]:
print(f'P(A) = {round(P1+P2+P3, 4)}')

P(A) = 0.3687


___

#### Ответ:

$P(A)\approx0.3687\approx36.87\%$

___

**3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:**  
   **a) первым спортсменом;**  
   **б) вторым спортсменом;**  
   **в) третьим спортсменом.**

Варианты развития событий:

$A$ - попадание в мишень
$B_1$ - попадание 1-м спортсменом
$B_2$ - попадание 2-м спортсменом
$B_3$ - попадание 3-м спортсменом

События $B_1, B_2, B_3$ состовляют полную группу событий:

$P(B_1) + P(B_2) + P(B_3) = \frac{1}{3}$

Событие $A$ при условии наступления $B_n$, даны в условии задачи:

$P_{B_1}(A) = 0.9$

$P_{B_2}(A) = 0.8$ 

$P_{B_3}(A) = 0.6$

По формуле полной вероятности, вероятность наступления события $A$:

$A$ - $P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$  

In [53]:
P_B = 1/3
P_A = P_B * 0.9 + P_B * 0.8 + P_B * 0.6
print(f'Р(А) = {P_A: .4f}')

Р(А) =  0.7667


Так как, событие $A$ уже произошло, с помощью формулы Байеса оценим вероятность того, что имело место одно из событий $B_n$  

$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}$

$\Downarrow\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}\\P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$  


In [63]:
P_A_B1 = P_B * 0.9/P_A
P_A_B2 = P_B * 0.8/P_A
P_A_B3 = P_B * 0.6/P_A
print(f'P_A_B1: {P_A_B1: .4f}')
print(f'P_A_B2: {P_A_B2: .4f}')
print(f'P_A_B3: {P_A_B3: .4f}')

P_A_B1:  0.3913
P_A_B2:  0.3478
P_A_B3:  0.2609


---
#### Ответ:  
а) Вероятность того, что выстрел произвёл первый спортсмен:  $P_{A}(B_1)\approx0.3913\approx39.13\%$;  
б) Вероятность того, что выстрел произвёл второй спортсмен:  $P_{A}(B_2)\approx0.3478\approx34.78\%$;  
в) Вероятность того, что выстрел произвёл третий спортсмен:  $P_{A}(B_3)\approx0.2609\approx26.09\%$.    

---